chain/chain/src/resharding.rs

```diff
@@ -305,7 +326,7 @@ impl Chain {
         let block_header = self.get_block_header(sync_hash)?;
         let prev_hash = block_header.prev_hash();
 
-        let child_shard_uids = state_roots.keys().collect_vec();
+        let child_shard_uids = state_roots.keys().cloned().collect_vec();
```

Comment:
Why do we need the clone here?

Reply:
This variable needs to outlive the state_roots
- state_roots are consumed in line 333
- child_shard_uids are needed in line 341
It's very small so I'm not worried about perf.